### PR TITLE
CIP-0036 | Change reward address field for Fund 10

### PR DESCRIPTION
**Pull request by Ryun1**
### Proposed Change
Replace `reward_address` with `payment_address`. Where `payment_address` should be a Shelley payment address. This means that the payment of voting rewards in Catalyst can no longer be done via MIR transfer but rather require regular transactions.

### Motivation
Past Catalyst's Fund 9, MIR transfers will be unavailable to be used by Project Catalyst. The operators of Project Catalyst, IOG, do not control the keys necessary to perform MIR transfers. This aligns Catalyst with all other projects implementing this standard.

Historically the use of MIR transfers by Project Catalyst was allowed to save on the costs associated with the payment of thousands of voting rewards. But this application was beyond the intended design of the mechanism since Project Catalyst is not a core network function.

The removal of this constraint makes the application of this standard more practical for projects beyond Catalyst to adopt.

**Note:** In an effort to ease the transition and not remove voting rights unexpectedly from already registered voters, Catalyst's Fund 10 registrations/delegations with the incorrect type of `reward_address` will still be valid, but will not be eligible to receive voting rewards.

cc: @kevinhammond @stevenj @ehanoc

### CODE REVIEW

**Comment by gitmachtl** [comment]
> Registrations/delegations with the incorrect type of `rewards_address` will still be valid, but will not receive voting rewards.

Why should an incorrect type still be valid?

**Comment by Ryun1** [comment]
> Why should an incorrect type still be valid?

Its a lesser sin to not get a reward for voting than be denied your voting rights, due to a reward address change outside of your control, that you may have not found out about until after its too late. 

Registrations should entitle you to vote. IF the rewards address is something we can pay to, AND you earn a reward, then a reward gets paid to it.

File: CIP-0036/test-vector.md
Comment (gitmachtl) [inline CIP-0036/test-vector.md:51]: typo -> bech32

File: CIP-0036/test-vector.md
Comment (gitmachtl) [inline CIP-0036/test-vector.md:58]: typo -> bech32

File: CIP-0036/test-vector.md
Comment (gitmachtl) [inline CIP-0036/test-vector.md:131]: the provided example is not valid. because the vote-public-key is actually the stake-public-key. as we derive the vote-public-key now from path 1694H/1815H/0H/0/0, it can't be the same as the stake-public-key.

maybe also use an example like:
```json
{
  "type": "PaymentVerificationKeyShelley_ed25519",
  "description": "Governance Verification Key",
  "cborHex": "5820423fa841abf9f7fa8dfa10dacdb6737b27fdb0d9bcd9b95d48cabb53047ab769"
}
```
bech32: "gov_vk1ggl6ssdtl8ml4r06zrdvmdnn0vnlmvxehnvmjh2ge2a4xpr6ka5suatdx2"

for the vote-public-key ?

File: CIP-0036/test-vector.md
Comment (Ryun1) [reply CIP-0036/test-vector.md:51]: Good spot :)

File: CIP-0036/test-vector.md
Comment (Ryun1) [reply CIP-0036/test-vector.md:58]: Good spot :)

File: CIP-0036/test-vector.md
Comment (Ryun1) [reply CIP-0036/test-vector.md:136]: Good spot - I will fix this

File: CIP-0036/test-vector.md
Comment (Ryun1) [reply CIP-0036/test-vector.md:131]: > The catalyst private signing key is a bit misleading without the public key

I had just reused the voting private key from previous test vectors, good idea to add in the public side.

> i am not sure why the payment vkey/skey at the top are included. What are they there for?

As for vkey/skey, you're right there are not needed, I was leaning towards including everything to aid transparency and recreation - this was also done in the [CIP-62](https://github.com/cardano-foundation/CIPs/pull/296) test vectors. I will ask other's opinions on this as I think this is a good point.

**Comment by gitmachtl** [comment]
lgtm

**Comment by GitCardano** [comment]
How are you going to explain this change to those of us who struggle with change?  I just figured out how to vote, Fund9 was my first.  What is "regular payment address"?  I hold my Cardano on Daedalus.

**Comment by gitmachtl** [comment]
> How are you going to explain this change to those of us who struggle with change? I just figured out how to vote, Fund9 was my first. What is "regular payment address"? I hold my Cardano on Daedalus.

don't worry. daedalus will get an update in time.

**Comment by Ryun1** [comment]
> How are you going to explain this change to those of us who struggle with change? I just figured out how to vote, Fund9 was my first. What is "regular payment address"? I hold my Cardano on Daedalus.

@GitCardano - This change is to be implemented by wallets and supporting tools. End users/voters should not have to do anything but keep their wallet/tool of choice updated.

File: CIP-0036/README.md
Comment (rphair) [inline CIP-0036/README.md:235]: Could we add a link to some Catalyst documentation that verifies this change?  If no link exists, could we add some background here about the Catalyst team's decision to do this?

**Review by rphair — COMMENTED** [review]
My reading of the dialogue so far is that all reservations have addressed so I would readily approve this, pending:
- @KtorZ - I assume will want to give it a good technical review (I'm not qualified to determine if there could be any hang-ups in doing it this way).
- @Ryun1 - Is there a way we can verify that IO > Catalyst team actually intends to do things this way? (see below)

File: CIP-0036/README.md
Comment (janmazak) [inline CIP-0036/README.md:29]:
```suggestion
3. Declares a payment address to receive Catalyst rewards
```

File: CIP-0036/README.md
Comment (janmazak) [inline CIP-0036/README.md:235]:
```suggestion
  - **Note:** up to Fund 9 `reward_address` was a Shelley reward address (a.k.a. stake address); from Fund 10 onwards, it will be a normal payment address.  This will allow rewards to be paid directly from the designated Catalyst pot rather than requiring MIR transfers.
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:13]: Can we go back to hex-only notation? Or, have a CIP eventually that standardize the cardano-cli text-envelope format. For now, this format has been breaking compatibility with itself every other release, so it's likely that these test vectors will be invalid in a month. 

In contrast, raw hex-encoded bytestring is forever-compatible and interoperable with anything 👍

File: CIP-0036/README.md
Comment (KtorZ) [inline CIP-0036/README.md:113]: That sentence reads funny now 🙃 ... "reward_address is a payment address". Naming things is hard I guess. 

-> Might be good to consider renaming the field altogether.

File: CIP-0036/README.md
Comment (KtorZ) [reply CIP-0036/README.md:235]: > This will allow rewards to be paid directly from the designated Catalyst pot

What is, the "designated Catalyst pot" ? I thought this pot was precisely called the "treasury" 😶 ?

File: CIP-0036/README.md
Comment (gitmachtl) [reply CIP-0036/README.md:113]: maybe we should change it here and in the cddl from `rewards_address` to `voting_rewards_payout_address`

File: CIP-0036/README.md
Comment (rphair) [reply CIP-0036/README.md]: @KtorZ from my reading of the CIP currently, Fund 10 will allow (or be based on) paying from accounts that are _not_ the treasury.  But @Ryun1 there is still a need to document this change.  The Catalyst changes are usually documented on a Git Doc somewhere & I still believe we should have a link to this resolution in the CIP text.

Comment (Ryun1) [reply CIP-0036/README.md]: @KtorZ @rphair 
Working on fleshing out the motivation for this.

File: CIP-0036/README.md
Comment (Ryun1) [reply CIP-0036/README.md:113]: I'm on the fence about renaming it, I would like further input.

Historically, before Catalyst's Fund 4, this field was a payment address and was named `rewards_address`, see #89 - it does not appear to have been an issue back then. With that said, I do understand additional the potential for confusion as this is not the first time this field has changed.

File: CIP-0036/test-vector.md
Comment (Ryun1) [reply CIP-0036/test-vector.md]: > Can we go back to hex-only notation?
Yes I agree - change made.

File: CIP-0036/README.md
Comment (stevenj) [reply CIP-0036/README.md]: @rphair I think there is a misunderstanding here.  Proj Catalyst did not make a decision to make this change.  Proj Catalyst was forced to by circumstance and directive.  IOG does not control the keys sufficient to perform MIR transfers, so its incapable of performing them.  The only documentation proj catalyst has on registration and delegation is CIP-15 and CIP-36.  The motivation therefore is "Project Catalyst would like to continue to have the option of paying voter rewards.  Now that MIR transfers are unavailable for that purpose, the only option is to change the reward address to a normal payment address.  This also has the added benefit that these formats can be utilized by other applications,  which is what the CIP-36 specifically envisioned when it added a voting purpose to the record.  In recognition that this change may cause a large number of previously registered voters to now have an 'invalid' registration,  the prudent course would seem to be not denying a voter the right to vote, simply because they would be unable to receive a voter reward through no fault of their own."

Comment (rphair) [reply CIP-0036/README.md]: thanks @stevenj, that was in fact what my understanding was... once we get confirmation @Ryun1 is done working on the Motivation as it appears in the CIP then I'll double check that all your points are clarified there.

File: CIP-0036/README.md
Comment (Ryun1) [reply CIP-0036/README.md]: @KtorZ @gitmachtl 

See most recent commit, I have renamed the field to just `payment_address`. In addition to making things clearer, I believe removing the notion of rewards from the field name makes the specification less Catalyst specific; as the notion of voting rewards are Catalyst specific.

File: CIP-0036/README.md
Comment (rphair) [reply CIP-0036/README.md]: thanks for updates @Ryun1 but so far I'm only seeing the Motivation section of the CIP draft has had "address" changed to "payment address" & nothing else.  @stevenj's [comments](https://github.com/cardano-foundation/CIPs/pull/373#discussion_r1061365560) are now reflected in the original post of this PR but I don't see them in the CIP document yet... have I missed something?

Comment (Ryun1) [reply CIP-0036/README.md]: @rphair 

Apologies I was unclear, poor choice of words - I was trying to say that @stevenj's comments have been reflected in this PR's motivation.

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:9]: There's an extra CBOR wrapping on this key. It should be:

```
614fdfe13d403bee2014570b190d81851f17d8daca0b6dd1ce33014403191003
```

(a private key is 32-byte long).

Comment (KtorZ) [inline CIP-0036/test-vector.md:9]:
```suggestion
614fdfe13d403bee2014570b190d81851f17d8daca0b6dd1ce33014403191003
```

There's an extra CBOR wrapping on this key which ought not to be there (a private key is 32-byte long).

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:14]:
```suggestion
7a24dd8e692cec94b612c2ec81f508aada96557c2052a447b9d197b006fa7d2a
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:19]:
```suggestion
852fa5d17df3efdfdcd6dac53ec9fe5593f3c0bd7cadb3c2af76c7e15dfa8a5c
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:12]:
```suggestion
Payment **public** verification hex:
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:7]:
```suggestion
Payment **private** signing key hex:
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:17]:
```suggestion
Staking **private** signing key hex:
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:22]:
```suggestion
Staking **public** verification key hex:
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:24]:
```suggestion
e3cd2404c84de65f96918f18d5b445bcb933a7cda18eeded7945dd191e432369
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:26]:
```suggestion
Governance (Catalyst) **public** verification key hex:
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:28]:
```suggestion
0036ef3e1f0d3f5989e2d155ea54bdb2a72c4c456ccb959af4c94868f473f5a0
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:31]:
```suggestion
Governance (Catalyst) **private** signing key hex:
```

File: CIP-0036/test-vector.md
Comment (KtorZ) [inline CIP-0036/test-vector.md:33]: Not sure what this is but it ain't a key? Could it be a combined private key +  public key ?

Comment (Ryun1) [reply CIP-0036/test-vector.md:33]: @KtorZ

Comment (KtorZ) [reply CIP-0036/test-vector.md:33]: Let's change `**private**` to `**extended private**` then 👍

**Comment by rphair** [comment]
@Ryun1 can you rebase & merge from upstream `master` to fix the current merge conflict (from https://github.com/cardano-foundation/CIPs/pull/426 being merged in the meantime)?  cc @KtorZ

**Review by rphair — APPROVED** [review]
Looks to me like last round of editing covers everything & I will leave it to @KtorZ & maybe also @SebastienGllmt to double-check.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**
